bleak/backends/winrt/scanner.py

```diff
@@ -0,0 +1,194 @@
+            ] = data[4:]
+        # 0x21 is service data with 128-bit UUID
+        for section in event_args.advertisement.get_sections_by_type(0x21):
+            reader = winstrm.DataReader.from_buffer(section.data)
+            data = [reader.read_byte() for _ in range(section.data.length)]
+            service_data[str(UUID(bytes=data[15::-1]))] = data[16:]
+
+        # Use the BLEDevice to populate all the fields for the advertisement data to return
+        advertisement_data = AdvertisementData(
+            local_name=event_args.advertisement.local_name,
+            manufacturer_data=device.metadata["manufacturer_data"],
+            service_data=service_data,
+            service_uuids=device.metadata["uuids"],
+            platform_data=(sender, event_args),
+        )
+
+        self._callback(device, advertisement_data)
+
+    def _stopped_handler(self, sender, e):
+        logger.debug(
+            "{0} devices found. Watcher status: {1}.".format(
+                len(self._devices), self.watcher.status
+            )
+        )
+
+    async def start(self):
+        self.watcher = winbleadv.BluetoothLEAdvertisementWatcher()
+        self.watcher.scanning_mode = self._scanning_mode
+
+        event_loop = asyncio.get_event_loop()
+
+        self._received_token = self.watcher.add_received(
+            lambda s, e: event_loop.call_soon_threadsafe(self._received_handler, s, e)
+        )
+        self._stopped_token = self.watcher.add_stopped(
+            lambda s, e: event_loop.call_soon_threadsafe(self._stopped_handler, s, e)
+        )
+
+        if self._signal_strength_filter is not None:
+            self.watcher.signal_strength_filter = self._signal_strength_filter
+        if self._advertisement_filter is not None:
+            self.watcher._advertisement_filter = self._advertisement_filter
+
+        self.watcher.start()
+
+    async def stop(self):
+        self.watcher.stop()
+
+        try:
+            self.watcher.remove_received(self._received_token)
+            self.watcher.remove_stopped(self._stopped_token)
+        except Exception as e:
+            logger.debug("Could not remove event handlers: {0}...".format(e))
+
+        self._stopped_token = None
+        self._received_token = None
+
+        self.watcher = None
+
+    async def set_scanning_filter(self, **kwargs):
+        pass
+
+    async def get_discovered_devices(self) -> List[BLEDevice]:
+        found = []
+        for event_args in list(self._devices.values()):
+            new_device = self.parse_eventargs(event_args)
+            if (
+                not new_device.name
+                and event_args.bluetooth_address in self._scan_responses
+            ):
+                new_device.name = self._scan_responses[
+                    event_args.bluetooth_address
+                ].advertisement.local_name
+            found.append(new_device)
+
+        return found
+
+    def parse_eventargs(self, event_args):
```

Comment:
maybe should be `_parse_event_args`?

Reply:
Yes. Made this change in the `dotnet` backend as well.